tests/Cucumber/CompatibilityTest.php

```diff
@@ -28,44 +28,75 @@
  * Tests the parser against the upstream cucumber/gherkin test data.
  *
  * @group cucumber-compatibility
+ *
+ * @phpstan-type TCucumberParsingTestCase array{mode: GherkinCompatibilityMode, file: SplFileInfo}
+ * @phpstan-type TKnownIncompatibilityMap array<value-of<GherkinCompatibilityMode>, array<string,string>>
  */
 class CompatibilityTest extends TestCase
 {
     private const GHERKIN_TESTDATA_PATH = __DIR__ . '/../../vendor/cucumber/gherkin-monorepo/testdata';
     private const EXTRA_TESTDATA_PATH = __DIR__ . '/extra_testdata';
 
     /**
-     * @var array<string, string>
+     * @phpstan-var TKnownIncompatibilityMap
      */
     private array $notParsingCorrectly = [
-        'complex_background.feature' => 'Rule keyword not supported',
-        'docstrings.feature' => 'Escaped delimiters in docstrings are not unescaped',
-        'datatables_with_new_lines.feature' => 'Escaped newlines in table cells are not unescaped',
-        'escaped_pipes.feature' => 'Escaped newlines in table cells are not unescaped',
-        'rule.feature' => 'Rule keyword not supported',
-        'rule_with_tag.feature' => 'Rule keyword not supported',
-        'tags.feature' => 'Rule keyword not supported',
-        'descriptions.feature' => 'Examples table descriptions not supported',
-        'descriptions_with_comments.feature' => 'Examples table descriptions not supported',
-        'feature_keyword_in_scenario_description.feature' => 'Scenario descriptions not supported',
-        'padded_example.feature' => 'Table padding is not trimmed as aggressively',
-        'spaces_in_language.feature' => 'Whitespace not supported around language selector',
-        'rule_without_name_and_description.feature' => 'Rule is wrongly parsed as Description',
-        'incomplete_background_2.feature' => 'Background descriptions not supported',
+        'legacy' => [
```

Comment:
It would be nice to directly use the enum case values as the array keys, but they're not supported in constant expressions in PHP 8.1.